src/server/main_service.cc

```diff
@@ -532,6 +532,10 @@ void Service::Init(util::AcceptServer* acceptor, util::ListenerInterface* main_i
   StringFamily::Init(&pp_);
   GenericFamily::Init(&pp_);
   server_family_.Init(acceptor, main_interface);
+
+  ChannelStore* cs = new ChannelStore{&channel_control_};
+  pp_.Await(
+      [cs](uint32_t index, ProactorBase* pb) { ServerState::tlocal()->UpdateChannelStore(cs); });
```

Comment:
I actually keep the control block as a main_service member, but it could be just a fully hidden static variable 🤔